### PR TITLE
Update plugin.py

### PR DESCRIPTION
**Pull request by artodrees**
Adds indirect parameterization support and fixture teardown grouping

### CODE REVIEW

**Review by jupe — COMMENTED** [review]
not yet working properly: 

A6354C1B-227A-4744-9B0C-0DE41A3FD35D

File: source/plugin.py [hunk 1]
```diff
@@ -109,4 +112,11 @@ def pytest_fixture_setup(request, fixturedef) -> None:
     # end of finalizers list (--> executed first)
     yield
 
-    pytest.grouping_session.end_github_group()
+    def _fixture_finalizer():
+        """
+        Start group "FIXTURE FixtureName TEARDOWN
+        """
+        pytest.grouping_session.start_github_group(prefix=fixture_type, name=fixture_name,
```
Comment (jupe) [inline source/plugin.py:130]: this probably leads to invalid `::endgroup::` 🤔

**Comment by hnrhonkala88** [comment]
fixed merge conflict when merged master to this PR and fixed bug that came up during the merge conflict fix.

**Comment by jupe** [comment]
still not handle lines properly:
image

**Comment by artodrees** [comment]
> still not handle lines properly: 
``

Yes, it's a pain with terminalreporter... 
i'm working on this atm.

**Comment by artodrees** [comment]
finally starting to look correct:
image

**Pull request merged**